### ISSUE TITLE
Is it possible to monitor socketio application by events

### ISSUE THREAD

**Issue by sparkingdark** [rycus86/prometheus_flask_exporter#141]
Hello,

Currently I am using python socketio for creating socket app , it it possible to monitor the app using this.

**Comment by rycus86** [comment]
Hi,

Does your app use Flask at all? This library is for easy integration with Flask, otherwise you might want to check out the underlying Prometheus client library at https://github.com/prometheus/client_python

**Comment by sparkingdark** [comment]
Yes my app use flask and other library which wrap it is the https://github.com/miguelgrinberg/Flask-SocketIO
Library so I am thinking to use this.

**Comment by rycus86** [comment]
Excellent, thanks! I wasn't aware of that integration.
You should be able to use this library then, which should give you the Prometheus client integration and basic HTTP metrics.
For the socketio managed methods you can try the metric helper functions (like `metrics.counter`, `metrics.summary`, etc) which should work I think, but admittedly haven't tried.
If you do try it I'd be interested to learn here how did you go with it?

**Comment by sparkingdark** [comment]
Yeah sure I am trying understand how the socketio and underlying technique work so far I got that basically this integration create a http long polling connection after that it transmit the events and get data from clientside.

I will update this issue after sometime I get or anykind of discussion related to this. Possibly I am also thinking to contribute the methods for only socketio apps or examples.

**Comment by sparkingdark** [comment]
So far I created a histogram and counter object with listening and receiving event label.Then I created a decorator function which I put on the top the listening event every-time that function receive some data it count and give the observed events. Now it's not the most efficient way but it gives the result back.
